### ISSUE TITLE
Wanted: Enable generated service

### ISSUE THREAD

**Issue by alexlarsson** [systemd/systemd#28006]
### Component

systemd

### Is your feature request related to a problem? Please describe

Podman has a systemd generator (called quadlet), basically you have a file `/etc/containers/systemd/minimal.container`:

```
[Unit]
Description=A minimal container
Before=local-fs.target

[Container]
Image=fedora
Exec=sleep 60
```

And it generates `/run/systemd/generator/minimal.service` which has `ExecStart=/usr/bin/podman run ...`.

If I try to enable the generated service, i get:

```
$ systemctl enable minimal.service 
Failed to enable unit: Unit /run/systemd/generator/minimal.service is transient or generated.
```

To work around this, the quadlet re-implements enable, so If you have the right `[Install]` section, the generator will create symlinks like `/run/systemd/generator/default.target.wants/minimal.service -> ../minimal.service`.

This gives you two options for shipping a quadlet file:  

 * Have an install section. This makes it super easy to start the app by default, but hard to not start it.
 * Don't have an install section. This makes it safe to install the file (it won't run anything by default), but it is a pain to enable the section.

### Describe the solution you'd like

What I would like it to be able to somehow use `systemctl enable` on the generated file and have it do the right thing.

**Comment by kdh8219** [comment]
Any update?

**Comment by richm** [comment]
> Any update?

No, but have you considered using the podman system role quadlet support?  https://github.com/linux-system-roles/podman/?tab=readme-ov-file#podman_quadlet_specs

**Comment by richm** [comment]
@kdh8219 is this a problem for other types of generated services other than podman?

**Comment by fynnay** [comment]
> [@kdh8219](https://github.com/kdh8219) is this a problem for other types of generated services other than podman?

Not a service, but I'm running into this when generating `.automount` units via fstab.

My scenario:

- I've got this entry in `/etc/fstab`:
  - `//server/share /mnt/server/share cifs _netdev,x-systemd.automount,x-systemd.mount-timeout=5,credentials=/mnt/server/.smbcredentials,uid=1000,gid=1000,iocharset=utf8 0 0`
- A reboot or `sudo systemctl daemon-reload` generates these files:
  - `/run/systemd/generator/mnt-server-share.mount`
  - `/run/systemd/generator/mnt-server-share.automount`
- ✔️ Running `sudo systemctl start mnt-server-share.automount` works as expected: The share is mounted on access
- ❌ Running `sudo systemctl enable mnt-server-share.automount` returns: `/run/systemd/generator/mnt-server-share.automount is transient or generated`
